packages/charts/src/chart_types/metric/renderer/dom/sparkline.tsx

```diff
@@ -21,6 +21,35 @@ export const getSparkLineColor = (color: MetricWTrend['color']) => {
   return hslToColor(h, s, l >= 0.8 ? l - 0.1 : l + 0.1, a);
 };
 
+/**
+ * Aligns and implicitly stacks all histogram trend data.
+ *
+ * Sometimes the trend data arrives with multiple series
+ * appended one on top of each other.
+ *
+ * @internal
+ */
+export const getSafeTrendData = (trend: MetricWTrend['trend']) => {
+  let lastX = trend[0]?.x;
+  const shouldBeSorted =
+    lastX !== undefined &&
+    trend.some(({ x }) => {
+      if (lastX === undefined) {
+        return false;
+      }
+      if (lastX > x) {
+        return true;
+      }
+      lastX = x;
+    });
+  if (!shouldBeSorted) {
+    return trend;
+  }
```

Comment:
```suggestion
  const shouldBeSorted = trend.some(({x}, index, array) => {
    if(index===0) return false;
    if(x < array[index-1]?.x){
        return true;
    }
    return false;
});
```
Can we use something like that (just pseudocode not checked)? To simplify a bit the code and also to avoid mutating the lastX variable from within the .some loop?